### DISCUSSION TITLE
Slint-lsp format command not visible

### DISCUSSION THREAD

**Discussion by duvenagep** [slint-ui/slint#4746] [Q&A] [answered]
Hi 

I am trying to use the slint-lsp format command on MacOS. All the lsp features except for the format is working.
I can see from the lsp-main.rs that the Format sub command  is present in the Clap CLI, however when I try to use it it does not appear to exist in the executable. I know it was moved 3 weeks ago in the [PR](https://github.com/slint-ui/slint/commit/f999f17421f86209c58c19a9826d40708f1e990):
```rust

#[derive(Clone, clap::Parser)]
#[command(author, version, about, long_about = None)]
pub struct Cli {
    #[arg(
        short = 'I',
        name = "Add include paths for the import statements",
        number_of_values = 1,
        action
    )]
    include_paths: Vec<std::path::PathBuf>,

    /// The style name for the preview ('native' or 'fluent')
    #[arg(long, name = "style name", default_value_t, action)]
    style: String,

    /// The backend or renderer used for the preview ('qt', 'femtovg', 'skia' or 'software')
    #[arg(long, name = "backend", default_value_t, action)]
    backend: String,

    /// Start the preview in full screen mode
    #[arg(long, action)]
    fullscreen: bool,

    /// Hide the preview toolbar
    #[arg(long, action)]
    no_toolbar: bool,

    #[command(subcommand)]
    command: Option<Commands>,
}

#[derive(Subcommand, Clone)]
enum Commands {
    /// Format slint files
    Format(Format),
}

#[derive(Args, Clone)]
struct Format {
    #[arg(name = "path to .slint file(s)", action)]
    paths: Vec<std::path::PathBuf>,

    /// modify the file inline instead of printing to stdout
    #[arg(short, long, action)]
    inline: bool,
}

```

Below a screenshot from the output of trying to use the slint-lsp
![Screenshot 2024-03-01 at 22 11 59](https://github.com/slint-ui/slint/assets/13404204/9f3b95c7-a487-4c99-bf64-ee51f6dc3deb)

Any idea what I am doing wrong?

**Comment by tronical** [comment, marked as answer]
You're not doing anything wrong per-se, it's just that the changes that merged this into the `slint-lsp` command are not in a release yet. So `cargo install slint-lsp` won't show it.

If you wanted, you could run `cargo install --git https://github.com/slint-ui/slint slint-lsp` - I think that should work.

**Reply by duvenagep** [reply]
Thanks! Will do in the mean time. Any idea when the new release will be published?

**Reply by tronical** [reply]
We're aiming for a release in the coming weeks, in March.

**Reply by duvenagep** [reply]
Clear, thanks!
